internal/git/cli.go

```diff
@@ -16,30 +17,55 @@ import (
 
 // GitCLI maintains a context for interacting with the Git command line executable.
 type GitCLI struct {
-	ctx   context.Context
-	exe   string
-	env   map[string]string
-	cwd   string
-	quiet bool
-	log   bool
+	ctx         context.Context
+	exe         string
+	env         map[string]string
+	cwd         string
+	quiet       bool
+	log         bool
+	mergeBinary string
 }
 
 // NewGitCLI creates a new GitCLI instance
 func NewGitCLI(ctx context.Context) (*GitCLI, error) {
-	exe, err := exec.LookPath("git")
-	if err != nil && !errors.Is(err, exec.ErrDot) {
-		return nil, err
-	} else if errors.Is(err, exec.ErrDot) {
-		if exe, err = filepath.Abs(exe); err != nil {
+	var exe string
+	var err error
+
+	// Check if the git CLI is available in the runner's temp directory
+	if gitCliPath := lookupGitCliInRunnerTemp(); gitCliPath != "" {
+		exe = gitCliPath
+	} else {
+		exe, err = exec.LookPath("git")
+		if err != nil && !errors.Is(err, exec.ErrDot) {
 			return nil, err
+		} else if errors.Is(err, exec.ErrDot) {
+			if exe, err = filepath.Abs(exe); err != nil {
+				return nil, err
+			}
 		}
 	}
+
+	mergeBinary, _ := exec.LookPath("merge")
 	cwd, err := os.Getwd()
 	if err != nil {
 		return nil, err
 	}
 	env := os.Environ()
-	return &GitCLI{ctx: ctx, exe: exe, env: envEntriesToMap(env), cwd: cwd, quiet: false, log: true}, nil
+	return &GitCLI{ctx: ctx, exe: exe, env: envEntriesToMap(env), cwd: cwd, quiet: false, log: true, mergeBinary: mergeBinary}, nil
+}
+
+func lookupGitCliInRunnerTemp() string {
```

Comment:
Why? We run the action in a container with git on the PATH. Why do we need to look up an alternative executable?

Reply:
After discussion this change has been rolled back to allow the `checkout` action to use it's own binary and the `merge` binary will use the one copied over by the DSL runtime.